mirdata/irmas.py

```diff
@@ -0,0 +1,406 @@
+                            os.path.join(data_home, directory, directory_)
+                        ):
+                            for file in files__:
+                                if file.endswith('.wav'):
+                                    test_index = str(count)
+                                    irmas_dict[test_index] = {
+                                        'genre': None,
+                                        'drum': None,
+                                        'train': False,
+                                    }
+                                    count += 1
+
+    irmas_dict["data_home"] = data_home
+
+    return irmas_dict
+
+
+DATA = utils.LargeData(
+    'irmas_index.json', _load_metadata
+)
+
+
+class Track(track.Track):
+    """IRMAS track class
+
+    Args:
+        track_id (str): track id of the track
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets/Mridangam-Stroke`
+
+    Attributes:
+        track_id (str): track id
+        train (bool): flag to identify if the track is from the training of the testing dataset
+        genre (str): string containing the namecode of the genre of the track.
+        drum (bool): flag to identify if the track contains drums or not.
+    """
+
+    def __init__(self, track_id, data_home=None):
+        if track_id not in DATA.index:
+            raise ValueError('{} is not a valid track ID in Example'.format(track_id))
+
+        self.track_id = track_id
+
+        if data_home is None:
+            data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+        self._data_home = data_home
+
+        self._track_paths = DATA.index[track_id]
+        self.audio_path = os.path.join(self._data_home, self._track_paths['audio'][0])
+
+        metadata = DATA.metadata(data_home)
+        if metadata is not None and track_id in metadata:
+            self._track_metadata = metadata[track_id]
+        else:
+            self._track_metadata = {'genre': None, 'drum': None, 'train': None}
+
+        self.annotation_path = os.path.join(
+            self._data_home, self._track_paths['annotation'][0]
+        )
+        self.genre = self._track_metadata['genre']
+        self.drum = self._track_metadata['drum']
+        self.train = self._track_metadata['train']
+
+    @utils.cached_property
+    def predominant_instrument(self):
+        """EventData: predominant instrument"""
+        return load_pred_inst(self.audio_path, self.annotation_path, self.train)
+
+    @property
+    def audio(self):
+        """(np.ndarray, float): audio signal, sample rate"""
+        return load_audio(self.audio_path)
+
+    def to_jams(self):
+        """Jams: the track's data in jams format"""
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            event_data=[(self.predominant_instrument, 'predominant_instrument')],
```

Comment:
I'm not too familiar with this dataset, but I'm curious about why this annotations are `eventdata`?

Reply:
See comment below